File: BE/hieuthuoc/src/main/java/com/example/hieuthuoc/entity/TonKho.java
package com.example.hieuthuoc.entity;

import java.util.Date;

import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import jakarta.persistence.Entity;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@Entity
@Table(name = "ton_kho")
@EntityListeners(AuditingEntityListener.class)
@EqualsAndHashCode(callSuper = true)
public class TonKho extends TimeAuditable{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "thuoc_id")
    private Thuoc thuoc;

    private String soLo;
    private Date hanSuDung;
    private Integer soLuong;
    private String viTri;

    // getters and setters
}

File: front/src/app/_model/tonkho.ts
import { PhieuNhap } from "./phieunhap";
import { Thuoc } from "./thuoc";

export class TonKho {
  id?: string;
  thuocId?: Thuoc;
  soLo?: String;
  hanSuDUng?: PhieuNhap;
  soLuong?: PhieuNhap;
  viTri?: String;
}
